### ISSUE TITLE
(0.35.x): Query With Same Column Names Doesn't Display Correctly

### ISSUE THREAD

**Issue by oliverisaac** [metabase/metabase#12252]
**Describe the bug**
When running a query with a join statement containing two column names, the value of the first column overwrites the value of the second column.

**Logs**
No relevant logs

**To Reproduce**
1. Create a new custom query question
2. Do a question query which generates a result containing two columns with the same name:
```sql
select * from (select 1 as id, 2 as joinon) s1 join (select 100 as id, 2 as joinon) s2 on s1.joinon = s2.joinon
```
3. The resulting table will return:

| id | joinon | id | joinon |
|----|--------|----|--------|
| 1  | 2      | 1  | 2      |

**Expected behavior**
The query should return:

| id | joinon | id  | joinon |
|----|--------|-----|--------|
| 1  | 2      | 100 | 2      |

In previous versons, metabase would return:

| id | joinon | id_2  | joinon_2 |
|----|--------|-----|--------|
| 1  | 2      | 100 | 2      |

**Screenshots**
<img width="1006" alt="Screen Shot 2020-04-02 at 5 22 47 PM" src="https://user-images.githubusercontent.com/28768356/78305512-ab27fe80-7506-11ea-95a0-0190381e9a20.png">

**Information about your Metabase Installation:**

You can get this information by going to Admin -> Troubleshooting.

- Your browser and the version: Chrome 80
- Your operating system: Mac OS
- Your databases: MariaDB
- Metabase version: 0.35.0, 0.35.1
- Metabase hosting environment: GKE
- Metabase internal database: MariaDB

**Severity**
How severe an issue is this bug to you? Is this annoying, blocking some users, blocking an upgrade or blocking your usage of Metabase entirely?

We had to downgrade to 0.34.3 as this breaks queries for us.

**Comment by flamber** [comment]
I'm unsure if this is a BE or FE issue. MBQL is not adding the `..._2` to the `display_name` anymore, but the data (`100`) is returned, and displaying it as a Bar actually shows the 100 column.

**Comment by tlrobinson** [comment]
Reproduced using `select 1 as foo, 2 as foo`.

It appears `field_ref` for the second (`FOO_2`) column is incorrect. It should be `["field-literal", "FOO_2", "type/Integer"]` not  `["field-literal", "FOO", "type/Integer"]`. This is a backend issue.

```json
{
  "data": {
    "rows": [[1, 2]],
    "cols": [
      {
        "display_name": "FOO",
        "base_type": "type/Integer",
        "source": "native",
        "field_ref": ["field-literal", "FOO", "type/Integer"],
        "name": "FOO"
      },
      {
        "display_name": "FOO",
        "base_type": "type/Integer",
        "source": "native",
        "field_ref": ["field-literal", "FOO", "type/Integer"],
        "name": "FOO_2"
      }
    ],
    "native_form": { "query": "select 1 as foo, 2 as foo\n" },
    "results_timezone": "GMT",
    "requested_timezone": "US/Pacific",
    "results_metadata": {
      "checksum": "ixc3o05OMjKEuT1L8n2+8A==",
      "columns": [
        {
          "display_name": "FOO",
          "name": "FOO",
          "base_type": "type/Integer",
          "special_type": null,
          "fingerprint": {
            "global": { "distinct-count": 1, "nil%": 0 },
            "type": { "type/Number": { "min": 1, "max": 1, "avg": 1, "sd": null, "q1": 1, "q3": 1 } }
          }
        },
        {
          "display_name": "FOO",
          "name": "FOO",
          "base_type": "type/Integer",
          "special_type": null,
          "fingerprint": {
            "global": { "distinct-count": 1, "nil%": 0 },
            "type": { "type/Number": { "min": 2, "max": 2, "avg": 2, "sd": null, "q1": 2, "q3": 2 } }
          }
        }
      ]
    },
    "insights": null
  },
  "database_id": 1,
  "started_at": "2020-04-03T19:30:51.84914Z",
  "json_query": {
    "database": 1,
    "native": { "template-tags": { }, "query": "select 1 as foo, 2 as foo\n" },
    "type": "native",
    "middleware": { "add-default-userland-constraints?": true }
  },
  "average_execution_time": null,
  "status": "completed",
  "context": "ad-hoc",
  "row_count": 1,
  "running_time": 22
}
```

**Comment by sbelak** [comment]
Fixed via #12345 and #12328.
